app/models/plant.rb

```diff
@@ -6,6 +6,11 @@ class Plant < ApplicationRecord
   has_one :watering_reminder, dependent: :destroy
   has_one_attached :plant_image, dependent: :destroy
 
+  enum light_requirements: { low_light: 0, medium_light: 1, high_light: 2 }
+  enum water_needs: { low_water: 0, medium_water: 1, high_water: 2 }
+  enum toxicity: { non_toxic: 0, to_humans: 1, to_pets: 2 }
```

Comment:
Two suggestions:
1. I see you had to add `_light/_water` suffixes to avoid conflicting names. You can simplify this by passing `_suffix: true` option instead (this will generate methods like `low_light_requirements?`, `low_water_needs?` etc)
2. you don't need to use hashes in this case, you can just pass an array of possible values and rails will automatically assign numbers to them. 
```suggestion
  enum light_requirements: [:low, :medium, :high], _suffix: true
  enum water_needs: [:low, :medium, :high], _suffix: true
  enum toxicity: [:non_toxic, :to_humans, :to_pets]
```

Reply:
But using hashes has some advantages 👍  especially if you decide to remove one of the values later. 

e.g. for `enum toxicity: [:non_toxic, :to_humans, :to_pets]` non_toxic will be saved in the DB as 0, to_humans as 1 and to_pets as 2.

If you one of the values later and change it to:
`enum toxicity: [:non_toxic, :to_pets]`
then rails will start interpreting 0 as non_toxic and 1 as to_pets (which is incorrect) and will probably error out when the value is 2.

With has you don't have this problem because you explicitly tell rails how to translate each value:
```
enum toxicity: { non_toxic: 0, to_pets: 2 }
```

But tbh it is best not to use integers for enums at all and always use string type for columns. If you access the database directly then you will have a hard time understanding what 0, 1 and 2 mean. 
To use string you need to have these columns created as strings and use 
```
enum toxicity: {non_toxic: "non_toxic", to_humans: "to_humans", to_pets: "to_pets"}
```

No need to change anything, it is just worth to know :)